src/WidgetBase.ts

```diff
@@ -71,6 +69,23 @@ export function diffProperty(propertyName: string, diffType = DiffType.CUSTOM, d
 }
 
 /**
+ * Decorator that can be used to enable property diffing of all properties
+ */
+export function diffProperties(constructor: Function): void;
```

Comment:
I think it is unreasonable. It makes for some strange looking code, but it feels like strange to have users to use `()` in this one case, just because we made the DiffType optional. `@diffProperties` feels a lot more natural to me than `@diffProperties()`